test.sh

```diff
@@ -530,70 +566,138 @@ function test_make_run_tests() {
+   # that basic usage of background threads still works.
+   # shellcheck disable=SC2086
+   "$BINDIR"/unit/task_system_test $use_shmem --num-bg-threads 4 --num-memtable-bg-threads  2
+
+   # shellcheck disable=SC2086
+   "$BINDIR"/driver_test io_apis_test $use_shmem
 }
 
 # ##################################################################
 # Run mini-unit-tests that were excluded from bin/unit_test binary:
 # Explicitly run individual cases from specific slow running unit-tests,
 # where appropriate with a different test-configuration that has been
 # found to provide the required coverage.
+# Execute this set w/ and w/o the "--use-shmem" arg.
 # ##################################################################
 function run_slower_unit_tests() {
+    local use_shmem=$1
 
-    run_with_timing "Splinter inserts test" "$BINDIR"/unit/splinter_test test_inserts
+    local use_msg=
+    if [ "$use_shmem" != "" ]; then
+        use_msg="using shared memory"
+   fi
+
+    local msg="Splinter inserts test ${use_msg}"
+
+    # Allow $use_shmem to come w/o quotes. Otherwise for default execution, we
+    # end up with empty '' parameter, which causes the argument parsing routine
+    # in the program to cough-up an error.
+    # shellcheck disable=SC2086
+    run_with_timing "${msg}" "$BINDIR"/unit/splinter_test ${use_shmem} test_inserts
 
     # Use fewer rows for this case, to keep elapsed times of MSAN runs reasonable.
-    run_with_timing "Splinter lookups test" \
-        "$BINDIR"/unit/splinter_test --num-inserts 2000000 test_lookups
+    msg="Splinter lookups test ${use_msg}"
+    local n_mills=2
+    local num_rows=$((n_mills * 1000 * 1000))
+    # shellcheck disable=SC2086
+    run_with_timing "${msg}" \
+        "$BINDIR"/unit/splinter_test ${use_shmem} --num-inserts 2000000 test_lookups
+
+    msg="Splinter print diagnostics test ${use_msg}"
+    # shellcheck disable=SC2086
+    run_with_timing "${msg}" \
+        "$BINDIR"/unit/splinter_test ${use_shmem} test_splinter_print_diags
 
-    run_with_timing "Splinter print diagnostics test" \
-        "$BINDIR"/unit/splinter_test test_splinter_print_diags
+    # Test runs w/ default of 1M rows for --num-inserts
+    n_mills=1
+    local n_threads=8
+    msg="Large inserts stress test, ${n_mills}M rows, ${n_threads} threads ${use_msg}"
+    # shellcheck disable=SC2086
+    run_with_timing "${msg}" \
+        "$BINDIR"/unit/large_inserts_bugs_stress_test ${use_shmem} --num-threads ${n_threads}
+
+    # Test runs w/ more inserts and enable bg-threads
+    n_mills=2
+    local n_threads=8
+    msg="Large inserts stress test, ${n_mills}M rows, ${n_threads} threads, 7 bg threads ${use_msg}"
+    # shellcheck disable=SC2086
+    run_with_timing "${msg}" \
+        "$BINDIR"/unit/large_inserts_bugs_stress_test ${use_shmem} \
+                                                      --num-threads ${n_threads} \
+                                                      --num-normal-bg-threads 4 \
```

Comment:
Some slower-running unit-tests have also been enhanced to run with `--use-shmem` option.

Again, more testing coverage for shared-memory integration is ON by default in every CI run.